hcn/hcnnetwork.go

```diff
@@ -321,6 +321,18 @@ func GetNetworkByName(networkName string) (*HostComputeNetwork, error) {
 func (network *HostComputeNetwork) Create() (*HostComputeNetwork, error) {
 	logrus.Debugf("hcn::HostComputeNetwork::Create id=%s", network.Id)
 
+	for _, ipam := range network.Ipams {
+		for _, subnet := range ipam.Subnets {
+			if subnet.IpAddressPrefix != "" {
+				for _, route := range subnet.Routes {
+					if route.NextHop == "" {
+						return nil, fmt.Errorf("Network Create Error, subnet has address prefix but no gateway specified.")
+					}
```

Comment:
This check is okay, to ensure that next hop is mandatory.

To validate default gateway, we need to check if NextHop 0.0.0.0/0 is passed in

Reply:
not sure I understand

Reply:
fixed